### PR TITLE
Read from files

### PR DESCRIPTION
**Pull request by bgolding355**
Solves #13

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,10 +31,51 @@ jobs:
       - name: Checkout Repo
         uses: actions/checkout@v3
       - name: Secret Scanner
-        uses: secret-scanner/action@0.0.2
+        uses: secret-scanner/action@main #or latest release
```
Comment (jsoref) [inline README.md:34]: Best practice is in fact to do `@full-sha`: 
https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions

File: README.md [hunk 2]
```diff
@@ -30,10 +31,51 @@ jobs:
...
 
-You can also pass arguments to `detect-secrets` by using `detect-secret-additional-args`. For information on the arguments that you can pass, visit [Yelp/detect-secrets#filters](https://github.com/Yelp/detect-secrets#filters). For example:
+### Ignoring false positives
+Often you might have strings that the secret scanner determines are secrets, but are actually harmless. Examples of these might be:
+1. Docker tags
+2. Git commit SHA's
+3. Randomly generated base64 strings
+
+For these cases, it is useful to ignore certain files, lines, or "secrets". You can do this using the files:
+- `.github/actions/secret-scanner/excluded_files_regex.txt`
```
Comment (jsoref) [inline README.md:44]: I'd suggest this as the filename ("txt" doesn't tell anyone anything):
```suggestion
# File: .github/actions/secret-scanner/excluded_files.patterns
```

(And similar for the others)

File: README.md [hunk 3]
```diff
@@ -30,10 +31,51 @@ jobs:
...
+- `.github/actions/secret-scanner/excluded_secrets_regex.txt`
+- `.github/actions/secret-scanner/excluded_lines_regex.txt`
+
+While the path defaults to `.github/actions/secret-scanner`, you can change this with the input `exclude_files_path`. Blank lines, or lines starting with `#` will be ignored.
```
Comment (jsoref) [inline README.md:48]:
```suggestion
While the path defaults to `.github/actions/secret-scanner`, you can change this with the input `exclude_files_path`. Blank lines and lines starting with `#` will be ignored.
```

File: README.md [hunk 4]
```diff
@@ -30,10 +31,51 @@ jobs:
...
+
+#### How to use excluded_files_regex.txt
+On each line, write the regex for a filepath that you wish to ignore. For example:
+```
+# File: .github/actions/secret-scanner/excluded_files_regex.txt
+# Lines starting with the char '#' are ignored
+.*-sealed.json
```
Comment (jsoref) [inline README.md:55]:
```suggestion
.*-sealed\.json$
```

File: README.md [hunk 5]
```diff
@@ -30,10 +31,51 @@ jobs:
...
+\.github/actions/spelling/.*
```
Comment (jsoref) [inline README.md:56]:
```suggestion
\.github/actions/spelling/
```

File: README.md (hunk 1 above)
Comment (bgolding355) [reply README.md:34]: Noted. I will drop the commit which switches from `0.0.2` to main for now. Then after this is merged update the README to point to the new HEAD commit

File: README.md [hunk 6]
```diff
@@ -33,7 +34,48 @@ jobs:
...
+
+For these cases, it is useful to ignore certain files, lines, or "secrets". You can do this using the files:
+- `.github/actions/secret-scanner/excluded_files.patterns`
+- `.github/actions/secret-scanner/excluded_secrets.patterns`
+- `.github/actions/secret-scanner/excluded_lines.patterns`
+
+While the path defaults to `.github/actions/secret-scanner`, you can change this with the input `exclude_files_path`. Blank lines and lines starting with `#` will be ignored.
+
+#### How to use excluded_files.patterns
+On each line, write the regex for a filepath that you wish to ignore. For example:
```
Comment (jsoref) [inline README.md:51]:
```suggestion
On each line, write the regex for the path to the file to ignore. For example:
```

File: README.md [hunk 7]
```diff
@@ -48,7 +48,7 @@ For these cases, it is useful to ignore certain files, lines, or "secrets". You
 While the path defaults to `.github/actions/secret-scanner`, you can change this with the input `exclude_files_path`. Blank lines and lines starting with `#` will be ignored.
 
 #### How to use excluded_files.patterns
-On each line, write the regex for a filepath that you wish to ignore. For example:
+On each line, write the regex for the path to the file to ignore. For example:
```
Comment (jsoref) [inline README.md:51]: With this change, you can drop the item from expect 😉

File: README.md (hunk 1 above)
Comment (jsoref) [reply README.md]: I think instead of pointing to head you should have `main` list the SHA of the latest release.

Comment (jsoref) [reply README.md]: But we'll probably want a new release with these new features 🙂

**Review by jsoref — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (bgolding355) [reply README.md]: I just created a new release. https://github.com/secret-scanner/action/pull/16

I think this PR will be my probably be last contribution to this project for the next 2 weeks so I just did this manually for now, but If I start working on this a lot again I will try and write something to bump this automatically whenever a new tag is released

**Pull request merged**